server/src/controllers/user.controller.js

```diff
@@ -0,0 +1,20 @@
+import User from "../models/user.model.js";
+
+export async function checkRole(req, res) {
+  // grab currently logged in user's profile data from results
+  console.log(JSON.stringify(req.headers))
+  const email = req.headers["x-user-email"];
+  console.log("email:", email)
+  const user = await User.findOne({
+    email: email,
+  });
+  if (user) {
+    if (user.role === "admin") {
+      res.json(user.role);
+    }
+  }
+  else {
+    res.json({ role: "resident" })
+  }
+
```

Comment:
It's a good idea to wrap async methods in a try-catch to cleanly handle errors.

Lines 8-18 could be simplified to:

```
const user = await User.findOne({ email });

// error handling if user is null

res.status(200)
res.send(user.role)

Reply:
Ahh that makes sense! Will also make a note to do.